include/javabind/core.hpp

```diff
@@ -59,7 +59,7 @@ namespace javabind
         using native_type = NativeType;
         using java_type = JavaType;
 
-        static_assert(sizeof(native_type) == sizeof(java_type), "C++ and JNI types are expected to match in size.");
+        static_assert(sizeof(native_type) <= sizeof(java_type), "JNI type is expected to at least the size of the C++ type.");
```

Comment:
I would keep the original `static_assert`, and use the new assertion only if `JAVABIND_INTEGER_WIDENING_CONVERSION` is defined.